sqlalchemy_exasol/base.py

```diff
@@ -762,48 +762,50 @@ def pre_exec(self):
         Note: Parameter replacement is done for server versions < 4.1.8 or
               in case a delete query is executed.
         """
-        if self.isdelete or self.root_connection.dialect.server_version_info < (
-            4,
-            1,
-            8,
-        ):
-            db_query = self.unicode_statement
-            for i in range(1, len(self.parameters)):
-                db_query += ", (" + ", ".join(["?"] * len(self.parameters[i])) + ")"
-            for db_para in self.parameters:
-                for value in db_para:
-                    ident = "?"
-                    if value is None:
-                        db_query = db_query.replace(ident, "NULL", 1)
-                    elif isinstance(value, int):
-                        db_query = db_query.replace(ident, str(value), 1)
-                    elif isinstance(value, (float, Decimal)):
-                        db_query = db_query.replace(ident, str(float(value)), 1)
-                    elif isinstance(value, bool):
-                        db_query = db_query.replace(ident, "1" if value else "0", 1)
-                    elif isinstance(value, datetime):
-                        db_query = db_query.replace(
-                            ident,
-                            "to_timestamp('%s', 'YYYY-MM-DD HH24:MI:SS.FF6')"
-                            % value.strftime("%Y-%m-%d %H:%M:%S.%f"),
-                            1,
-                        )
-                    elif isinstance(value, date):
-                        db_query = db_query.replace(
-                            ident,
-                            "to_date('%s', 'YYYY-MM-DD')" % value.strftime("%Y-%m-%d"),
-                            1,
-                        )
-                    elif isinstance(value, bytes):
-                        db_query = db_query.replace(
-                            ident, "'%s'" % value.decode("UTF-8"), 1
-                        )
-                    elif isinstance(value, str):
-                        db_query = db_query.replace(ident, "'%s'" % value, 1)
-                    else:
-                        raise TypeError("Data type not supported: %s" % type(value))
-            self.statement = db_query
-            self.parameters = [[]]
+        server_version = self.root_connection.dialect.server_version_info
+        skip_pre_exec = (
+            not self.isdelete and server_version is None or server_version >= (4, 1, 8)
```

Comment:
I think, it is save to drop support for Exasol version 4

Reply:
nice, still we need it for the delete case I think (at least needs to be clarified).

Reply:
yep, think the same